dspace-api/src/main/java/org/dspace/authorize/service/AuthorizeService.java

```diff
@@ -604,4 +606,24 @@ long countAdminAuthorizedCollection(Context context, String query)
     public void replaceAllPolicies(Context context, DSpaceObject source, DSpaceObject dest)
             throws SQLException, AuthorizeException;
 
+    /**
+     * Finds all Indexed Objects where the current user has admin rights. If the user is an super Admin,
+     * this is all Indexed Objects(community/collection). Otherwise, it includes those objects where
+     * an indexed "admin" policy lists either the eperson or one of the eperson's groups
+     *
+     * @param context                    DSpace context
+     * @param discoverQuery
+     * @param entityType                 limit the returned object(community/collection) to those related to
+     * given entity type
+     * @param community                  parent community, could be null
+     * @param q                          limit the returned collection to those with metadata values matching the query
+     *                                   terms. The terms are used to make also a prefix query on SOLR
+     *                                   so it can be used to implement an autosuggest feature over the collection name
+     * @return                           DiscoveryResult object
+     * @throws SQLException              if something goes wrong
+     * @throws SearchServiceException    if search error
+     */
+    public DiscoverResult retrieveObjectWithAdmin(Context context, DiscoverQuery discoverQuery,
```

Comment:
This method seems similar to `AuthorizeService.findAdminAuthorizedCollection()` and `AuthorizeService.findAdminAuthorizedCommunity()`, both of which already exist.  

Is there a reason to create another method which is finding these same objects that you have admin permissions on?

I'm just a bit confused why this new method is being created to find the Communities & Collections that you have ADMIN rights on...we already seem to have methods to do that.